Source/WebKit/NetworkProcess/Downloads/PendingDownload.cpp

```diff
@@ -75,17 +75,31 @@ void PendingDownload::cancel(CompletionHandler<void(std::span<const uint8_t>)>&&
 }
 
 #if PLATFORM(COCOA)
+#if HAVE(MODERN_DOWNLOADPROGRESS)
+void PendingDownload::publishProgress(const URL& url, std::span<const uint8_t>&& bookmarkData)
+{
+    ASSERT(!m_progressURL.isValid());
+    m_progressURL = url;
+    m_bookmarkData = WTFMove(bookmarkData);
```

Comment:
Is this really safe? `m_bookmarkData` is a span, so we don't own the bytes. What makes sure the memory the span is pointed to in `m_bookmarkData` remains valid between this function call and the `didBecomeDownload` function call which ends up using the bookmark data?

Reply:
That is a great catch! I have changed the `m_bookmarkData` member to a Vector, and addressed further issues.

Thanks for reviewing!